packages/core/src/AttesterSchema.ts

```diff
@@ -0,0 +1,278 @@
+
+    private maxFieldSize: number
+
+    constructor() {
+        this.field = []
+        this.fieldOffset = []
+        this.currentFieldSize = BigInt(0)
+
+        this.maxFieldSize = 254
+    }
+
+    public addOrPass(entry: string, size: bigint): boolean {
+        if (this.currentFieldSize + size > this.maxFieldSize) {
+            return false
+        }
+        this.fieldOffset.push(Number(this.currentFieldSize))
+        this.currentFieldSize += size
+        this.field.push(entry)
+        return true
+    }
+
+    public getFieldOffset(entry: string): number {
+        for (let i = 0; i < this.field.length; i++) {
+            if (this.field[i] == entry) {
+                return this.fieldOffset[i]
+            }
+        }
+        return 0
+    }
+
+    public setMaxFieldSize(_maxFieldSize: number) {
+        this.maxFieldSize = _maxFieldSize
+    }
+}
+export const parseSchema = (schema: {}[]) => {
+    const SCHEMA_FIELDS = ['name', 'type', 'updateBy']
+
+    schema.forEach((entry) => {
+        if (!('name' in entry && 'type' in entry)) {
+            throw Error('Missing name or type in schema')
+        }
+        if (!('updateBy' in entry)) {
+            entry['updateBy'] = 'replace'
+        }
+
+        // Contains an invalid key
+        if (Object.keys(entry).some((x) => SCHEMA_FIELDS.indexOf(x) < 0)) {
+            throw Error('Schema includes an invalid key')
+        }
+
+        if (entry['type'] == 'uint') {
+            entry['type'] = 'uint253'
+        } else if (entry['type'] == 'bool') {
+            entry['type'] = 'uint1'
+        }
+
+        if (
+            String(entry['type']).slice(0, 4) != 'uint' ||
+            isNaN(+String(entry['type']).slice(4))
+        ) {
+            throw Error('Schema includes an invalid type')
+        }
+
+        if (entry['updateBy'] != 'sum' && entry['updateBy'] != 'replace') {
+            throw Error('Schema includes an invalid update method')
+        }
+    })
+    return schema
+}
+
+export class AttesterSchema {
+    private sumDataFields: DataField[]
+    private replDataFields: DataField[]
+
+    private curSumDataFieldCount: number
+    private curReplDataFieldCount: number
+    private schema: {}[]
+    private schemaLookup: {}
+
+    private encodedData: bigint[]
+
+    constructor(_schema: {}[]) {
+        this.curSumDataFieldCount = 0
+        this.curReplDataFieldCount = 0
+
+        this.sumDataFields = new Array(SUM_FIELD_COUNT)
+        this.replDataFields = new Array(FIELD_COUNT - SUM_FIELD_COUNT)
+
+        for (let i = 0; i < this.sumDataFields.length; i++) {
+            this.sumDataFields[i] = new DataField()
+        }
+
+        for (let i = 0; i < this.replDataFields.length; i++) {
+            this.replDataFields[i] = new DataField()
+        }
+
+        this.schema = parseSchema(_schema)
+        this.encodedData = new Array(FIELD_COUNT).fill(BigInt(2) ** BigInt(254))
+
+        this.encodedData = new Array(FIELD_COUNT).fill(BigInt(0))
```

Comment:
is the line 106 replaced by the line 108 ?

Reply:
Yep, I fixed it lol